lib/store/podDataSlice.ts

```diff
@@ -0,0 +1,51 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+import {
+    createPodDataFromAdapter,
+    PacketUpdate,
+    PodDataAdapter,
+} from "../adapters";
+import { PodData, updatePodData as updatePackets, Board } from "../models";
+import { create, StateCreator, StoreApi, UseBoundStore } from "zustand";
+
+export interface PodDataSlice {
+    podData: PodData
+    initPodData: (podDataAdapter: PodDataAdapter) => void
+    updatePodData: (newPodData: Record<number, PacketUpdate>) => void
+}
+
+export const podDataSlice: StateCreator<PodDataSlice> = (set, get) => ({
+    podData: {
+        boards: [] as Board[],
+        packetToBoard: {} as Record<number, number>,
+        lastUpdates: {} as Record<string, PacketUpdate>,
+    },
+
+    /**
+     * Reducer that initializes the state based on podDataAdapter.
+     * It uses a helper function createPodDataFromAdapter to do it.
+     * @param {PodDataAdapter} podDataAdapter 
+     */
+    initPodData: (podDataAdapter: PodDataAdapter) => {
+        set(state => ({
+            ...state,
+            podData: createPodDataFromAdapter(podDataAdapter)
+        }))
+    },
```

Comment:
make this function do what `createPodDataFromAdapter` does. This just adds an unnecessary extra layer of complexity and is really unlikely you will just use `createPodDataFromAdapter` to load it without loading the store